xdsl/traits.py

```diff
@@ -413,11 +413,22 @@ class CallableOpInterface(OpTrait, abc.ABC):
     """
 
     @classmethod
+    @abc.abstractmethod
     def get_callable_region(cls, op: Operation) -> Region:
         """
         Returns the body of the operation
         """
-        raise NotImplementedError
+        raise NotImplementedError()
+
+    @classmethod
+    @abc.abstractmethod
+    def get_argument_types(cls, op: Operation) -> tuple[Attribute, ...]:
+        raise NotImplementedError()
+
+    @classmethod
+    @abc.abstractmethod
+    def get_result_types(cls, op: Operation) -> tuple[Attribute, ...]:
+        raise NotImplementedError()
```

Comment:
Is there a reference you can give to the MLIR equivalent?

Reply:
I tried to hunt for this, but the best one I found was at the link in the docstring for the whole trait. I also found it quite difficult to navigate the tablegen.